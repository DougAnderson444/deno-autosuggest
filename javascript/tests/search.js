const mod = require('../build/index')


describe('Suggest', () => {
    beforeAll(() => {
        mod.storeRecords([
            {id: 10, text: 'Hello world!'},
            {id: 20, text: 'Foo bar'},
            {id: 30, text: '-BAZZZ-'},
        ])
    })
    
    test('Empty input', () => {
        const hits = mod.search('')
        expect(hits).toMatchSnapshot()
    })

    test('Equality', () => {
        const hits = mod.search('foo bar')
        expect(hits).toMatchSnapshot()
    })

    test('Prefix', () => {
        const hits = mod.search('ba')
        expect(hits).toMatchSnapshot()
    })
})