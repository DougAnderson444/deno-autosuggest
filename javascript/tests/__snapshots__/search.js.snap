// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest Empty input 1`] = `
Array [
  Object {
    "highlighted": "Hello world!",
    "id": 10,
    "text": "Hello world!",
  },
  Object {
    "highlighted": "Foo bar",
    "id": 20,
    "text": "Foo bar",
  },
  Object {
    "highlighted": "-BAZZZ-",
    "id": 30,
    "text": "-BAZZZ-",
  },
]
`;

exports[`Suggest Equality 1`] = `
Array [
  Object {
    "highlighted": "[Foo] [bar]",
    "id": 20,
    "text": "Foo bar",
  },
]
`;

exports[`Suggest Prefix 1`] = `
Array [
  Object {
    "highlighted": "Foo [ba]r",
    "id": 20,
    "text": "Foo bar",
  },
  Object {
    "highlighted": "-[BA]ZZZ-",
    "id": 30,
    "text": "-BAZZZ-",
  },
]
`;
